### PR TITLE
Upgrade `flappi` to Ruby 3.1.4

### PR DESCRIPTION
**Pull request by alto**
See [Asana ticket](https://app.asana.com/0/1202486819364117/1204427748739469) for details.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -12,6 +12,9 @@ Gemspec/RequiredRubyVersion:
   Enabled: false
 
 
+Layout/BlockAlignment:
```
Comment (alto) [inline .rubocop.yml:15]: This is a Ruby 3.1 feature only. Let's enable once we switched over from 3.0 .

File: flappi.gemspec [hunk 2]
```diff
@@ -19,7 +18,7 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'mocha'
   s.add_development_dependency 'pry-byebug'
   s.add_development_dependency 'rake', '>= 12.3'
-  s.add_development_dependency 'rubocop', '1.3.0'
+  s.add_development_dependency 'rubocop', '~> 1.14.0'
```
Comment (alto) [inline flappi.gemspec:21]: I needed this upgrade to enable `TargetRubyVersion: 3.1`.

File: lib/flappi/response_builder.rb [hunk 3]
```diff
@@ -266,7 +266,7 @@ def link(*link_params)
                    link_params.first
                  else
                    keys = %i[key path]
-                   Hash[link_params.each_with_index.map { |p, idx| [keys[idx], p] }]
+                   link_params.each_with_index.map { |p, idx| [keys[idx], p] }.to_h
```
Comment (fsanggang) [inline lib/flappi/response_builder.rb:269]: Change makes sense, but was there a new cop/rule that complained about this? (and the other changes in this PR)

File: lib/flappi.rb [hunk 4]
```diff
@@ -9,7 +9,7 @@
 root_path = "#{File.dirname(__FILE__)}/flappi/"
 dependent_file_paths = DEPENDENT_FILES.map { |f| root_path + f }
 
-found_file_paths = Dir["#{root_path}**/*.rb"].sort
+found_file_paths = Dir["#{root_path}**/*.rb"]
```
Comment (alto) [inline lib/flappi.rb:12]: Due to [Lint/RedundantDirGlobSort](https://docs.rubocop.org/rubocop/cops_lint.html#lintredundantdirglobsort).

File: system/hooks/rubocop_pre_commit.rb [hunk 5]
```diff
@@ -3,12 +3,12 @@
 require 'english'
 require 'rubocop'
 
-ADDED_OR_MODIFIED = /A|AM|^M/.freeze
+ADDED_OR_MODIFIED = /A|AM|^M/
```
Comment (alto) [inline system/hooks/rubocop_pre_commit.rb:6]: Due to [Style/RedundantFreeze](https://docs.rubocop.org/rubocop/cops_style.html#styleredundantfreeze).

File: system/hooks/rubocop_pre_commit.rb [hunk 6]
```diff
@@ -3,12 +3,12 @@
...
 FILE_EXTENSIONS_TO_CONSIDER = ['.rb', '.rake'].freeze
 
 changed_files = `git status --porcelain`.split(/\n/)
                                         .select { |file_name_with_status| file_name_with_status =~ ADDED_OR_MODIFIED }
-                                        .map { |file_name_with_status| file_name_with_status.split(' ')[1] }
+                                        .map { |file_name_with_status| file_name_with_status.split[1] }
```
Comment (alto) [inline system/hooks/rubocop_pre_commit.rb:11]: Due to [Style/RedundantArgument](https://docs.rubocop.org/rubocop/cops_style.html#styleredundantargument).

File: test/definition_locator_test.rb [hunk 7]
```diff
@@ -41,7 +41,8 @@ class ::Flappi::DefinitionLocatorTest < MiniTest::Test
         located_class = Flappi::DefinitionLocator.locate_class('DoesntExist', 'v2.0')
       end
 
-      assert_equal 'Endpoint DoesntExist is not defined to Flappi: Could not load Examples::DoesntExist because uninitialized constant Examples::DoesntExist was raised', ex.message
+      assert_match(/Endpoint DoesntExist is not defined to Flappi/, ex.message)
+      assert_match(/uninitialized constant Examples::DoesntExist/, ex.message)
```
Comment (alto) [inline test/definition_locator_test.rb:45]: Due to parsing error in rubocop.

File: lib/flappi/response_builder.rb (hunk 3 above)
Comment (alto) [reply lib/flappi/response_builder.rb:269]: Yep, due to [Style/HashConversion](https://docs.rubocop.org/rubocop/cops_style.html#stylehashconversion).

I have now marked all violated rubocop rules in this PR.

**Review by fsanggang — APPROVED** [review]

**Pull request merged**